scipy/sparse/compressed.py

```diff
@@ -1074,8 +1074,9 @@ def __get_has_canonical_format(self):
             # not sorted => not canonical
             self._has_canonical_format = False
         elif not hasattr(self, '_has_canonical_format'):
-            self.has_canonical_format = _sparsetools.csr_has_canonical_format(
-                len(self.indptr) - 1, self.indptr, self.indices)
+            self.has_canonical_format = bool(
```

Comment:
Looks like we're missing a leading underscore here.

Reply:
This also sets `has_sorted_indices` to `True` if `has_canonical_format` is `True`.

Reply:
Ah, right, property setters are subtle.